notes/advanced-data-types.md

```diff
@@ -0,0 +1,530 @@
+# Advanced Data Types
+
+## Introduction
+
+Here we will discuss types that are unique to C++. This includes `stucts`, `enums` and `unions`. We will also discuss `namespace`, which is not a type but rather an entity that holds variables and functions.
+
+## Structures (`structs`)
+
+A `struct` is an entities capable of holding fields, and you can think of it as a way to hold many variables of different types. 
+
+
+Below, we create a struct with the name `myBorland`, with fields `myField1` and `myField2`
+
+```
+/*
+struct {
```

Comment:
Maybe add a disclaimer here that the following format for making a struct is kinda unusual? I.e. you'd almost always want to do
`struct structName{...}`